code/datums/flock/flock.dm

```diff
@@ -541,33 +540,35 @@
 		if (RL_Started) RL_UPDATE_LIGHT(T)
 
 	for(var/obj/O in T)
-		if(istype(O, /obj/machinery/door/feather))
-			// repair door
-			var/obj/machinery/door/feather/door = O
-			door.heal_damage()
-			animate_flock_convert_complete(O)
 		for(var/keyPath in flock_conversion_paths)
-			var/obj/replacementPath = flock_conversion_paths[keyPath]
 			if(istype(O, keyPath))
-				if(isnull(replacementPath))
-					qdel(O)
-				else
-					var/dir = O.dir
-					var/obj/converted = new replacementPath(T)
-					// if the object is a closet, it might not have spawned its contents yet
-					// so force it to do that first
-					if(istype(O, /obj/storage))
-						var/obj/storage/S = O
-						if(!isnull(S.spawn_contents))
-							S.make_my_stuff()
-					// if the object has contents, move them over!!
-					for (var/obj/OO in O)
-						OO.set_loc(converted)
-					for (var/mob/M in O)
-						M.set_loc(converted)
-					qdel(O)
-					converted.set_dir(dir)
-					animate_flock_convert_complete(converted)
+				if (istype(O, /obj/machinery/door))
+					if (istype(O, /obj/machinery/door/firedoor/pyro))
+						qdel(O)
+						continue
+					if (istype(O, /obj/machinery/door/window) || istype(O, /obj/machinery/door/airlock/pyro/glass/windoor) || istype(O, /obj/machinery/door/unpowered/wood))
+						qdel(O)
+						continue
+					if (istype(O, /obj/machinery/door/poddoor/pyro) && !(findtext(O.name, "cargo") || findtext(O.name, "pod") || findtext(O.name, "mass") || findtext(O.name, "disposal"))) // shutters
+						qdel(O)
+						continue
```

Comment:
Maybe just allow `replacement_path` to be `null` and have that signify that that type is just deleted when its found?

Reply:
The issue with using null is that if a Flockdoor is over a tile that isn't a Flock floortile, and you convert that tile, the Flockdoor is deleted, meaning in your conversion list you have to say /obj/machinery/door/feather = /obj/machinery/door/feather, which seemed a bit off, and everything checked for here would also have to be put in the list with a null replacement, which I'm not sure if it's a good idea adding more checks considering its checking everything in the list for every object on a turf. Idk.